packages/x-date-pickers-pro/src/SingleInputDateRangeField/useSingleInputDateRangeField.ts

```diff
@@ -78,67 +79,53 @@ export const dateRangeFieldValueManager: FieldValueManager<
 
     return `${startDateStr}${endDateStr}`;
   },
-  getValueFromSections: (utils, prevSections, newSections, format) => {
-    const removeLastSeparator = (sections: DateRangeFieldSection[]) =>
-      sections.map((section, sectionIndex) => {
-        if (sectionIndex === sections.length - 1) {
-          return { ...section, separator: null };
-        }
-
-        return section;
-      });
-
-    const prevDateRangeSections = splitDateRangeSections(prevSections);
-    const dateRangeSections = splitDateRangeSections(newSections);
-
-    const startDateStr = createDateStrFromSections(
-      removeLastSeparator(dateRangeSections.startDate),
-    );
-    const endDateStr = createDateStrFromSections(dateRangeSections.endDate);
-
-    const startDate = utils.parse(startDateStr, format);
-    const endDate = utils.parse(endDateStr, format);
+  getValueFromSections: ({ utils, sections, format }) => {
+    const dateRangeSections = splitDateRangeSections(sections);
 
-    const shouldPublish =
-      (startDateStr !==
-        createDateStrFromSections(removeLastSeparator(prevDateRangeSections.startDate)) &&
-        utils.isValid(startDate)) ||
-      (endDateStr !== createDateStrFromSections(prevDateRangeSections.endDate) &&
-        utils.isValid(endDate));
+    const startDate = createDateFromSections({
+      utils,
+      sections: removeLastSeparator(dateRangeSections.startDate),
+      format,
+    });
+    const endDate = createDateFromSections({
+      utils,
+      sections: dateRangeSections.endDate,
+      format,
+    });
 
-    return {
-      value: [startDate, endDate] as DateRange<any>,
-      shouldPublish,
-    };
+    return [startDate, endDate] as DateRange<any>;
   },
-  getActiveDateFromActiveSection: (value, activeSection) => {
-    const updateActiveDate = (dateName: 'start' | 'end') => (newActiveDate: any) => {
-      if (dateName === 'start') {
-        return [newActiveDate, value[1]] as DateRange<any>;
-      }
+  getActiveDateFromActiveSection: ({ state, publishValue, activeSection }) => {
```

Comment:
That function is far from simple to understand. maybe commenting the returned object could help do understand its purpose

Reply:
I need to comment the whole `FieldValueManager` interface in depth